timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLockService.java

```diff
@@ -177,9 +179,11 @@ private AsyncResult<HeldLocks> acquireImmutableTimestampLock(UUID requestId, lon
     }
 
     private AsyncResult<HeldLocks> acquireNamedTimestampLockInternal(
-            TimestampLeaseName timestampName, UUID requestId, long timestamp) {
-        AsyncLock lock = lockManager.getNamedTimestampLock(timestampName, timestamp);
-        return lockAcquirer.acquireLocks(requestId, OrderedLocks.fromSingleLock(lock), TimeLimit.zero());
+            UUID requestId, Set<TimestampLeaseName> timestampNames, long timestamp) {
```

Comment:
This is a set because the ordering doesn't matter.

In HeldLocks, the list is just used to be iterated on in lock/unlock operations. And, the lock descriptors are returned as a set.

Reply:
Ah, looks like we thought about the same thing. I agree it doesn't matter, but it definitely reads weird. So maybe to allay these concerns, add an explicit comment OR handle the ordering.